crates/holochain_state/src/validation_receipts.rs

```diff
@@ -69,26 +71,27 @@ impl ValidationReceipt {
                 async move { validator.sign(&keystore, this).await }
             })
             .collect::<Vec<_>>();
-        let signatures = futures::stream::iter(futures)
-            .buffer_unordered(10)
-            .collect::<Vec<_>>()
-            .await;
-        let signatures = if signatures.iter().any(Result::is_ok) {
-            signatures.into_iter().filter_map(Result::ok).collect()
-        } else {
-            return Err(signatures
-                .into_iter()
-                .filter_map(Result::err)
-                .next()
-                .expect("Must contain at least one error if there is no success"));
-        };
+        let stream = futures::stream::iter(futures);
+        let signatures = try_stream_of_results(stream).await?;
+        if signatures.is_empty() {
+            return Ok(None);
+        }
```

Comment:
This is still weird, because this is exactly the case that should be impossible, yet you're letting it be `Ok`. I think we ultimately need to understand why a non-empty list of `validators` (due to line 57) could ever map to an empty stream of signatures. Shouldn't that be impossible? So, if we don't have any idea now (I don't), then I feel there needs to at least be logging here so we can see it happening and maybe get some clues from the real-world context.

Reply:
Ok I've added back in the expect because the tests shows that with the new `try_stream_of_results` an empty output is impossible if the input is empty.